napari/layers/utils/_link_layers.py

```diff
@@ -12,9 +13,30 @@
 
 #: Record of already linked layers... to avoid duplicating callbacks
 #  in the form of {(id(layer1), id(layer2), attribute_name) -> callback}
-LinkKey = Tuple[int, int, str]
+LinkKey = Tuple['ReferenceType[Layer]', 'ReferenceType[Layer]', str]
 Unlinker = Callable[[], None]
-_LINKED_LAYERS: dict[LinkKey, Unlinker] = dict()
+_UNLINKERS: dict[LinkKey, Unlinker] = dict()
+_LINKED_LAYERS: DefaultDict[
+    ReferenceType[Layer], Set[ReferenceType[Layer]]
+] = DefaultDict(set)
+
+
+def layer_is_linked(layer: Layer) -> bool:
+    """Return True if `layer` is linked to any other layers."""
+    return ref(layer) in _LINKED_LAYERS
+
+
+def get_linked_layers(*layers: Layer) -> Set[Layer]:
```

Comment:
this new `get_linked_layers` function that takes some layers and tells you what they are linked to.  it required changing the way the link-keys are stored from layer `ids` to layer `weakrefs` ... hence some of the other changes in this module.